### PR TITLE
Enhancement/9335 SiWG Setup Banner CTA

### PR DESCRIPTION
**Pull request by jimmymadon**
## Summary

Addresses issue:

- #9335 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.
- [x] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

**Review by zutigrm — APPROVED** [review]
Thanks @jimmymadon LGTM

Although I think refactoring the description paragraph could have been better as a separate issue, since it includes additional changes, they are fairly small so I think it is fine to be included as part of the PR

File: assets/js/components/consent-mode/ConsentModeSetupCTAWidget.js
Comment (tofumatt) [inline assets/js/components/consent-mode/ConsentModeSetupCTAWidget.js:1]: The consent mode changes in this PR are intentional, right? I don't see any notes about them in the PR description, but I'm guessing it's just some HTML element updates that were needed?

Just making sure they should be here…

File: assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js [hunk 1]
```diff
@@ -29,21 +29,26 @@ import CTALink from './CTALink';
...
-	);
+	const isNavigatingToCTALink = useSelect( ( select ) => {
+		return ctaLink
+			? select( CORE_LOCATION ).isNavigatingTo( ctaLink )
+			: false;
+	} );
 
 	return (
-		<div className="googlesitekit-publisher-win__actions">
+		<div className={ className }>
```
Comment (tofumatt) [inline assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js:46]: We shouldn't allow developers to _replace_ the existing class name, I'd think… so we should do this instead:

```suggestion
		<div className={ classnames( 'googlesitekit-publisher-win__actions', className ) }>
```

File: assets/js/googlesitekit/notifications/components/common/Description.js [hunk 2]
```diff
@@ -19,9 +19,13 @@
...
 
-export default function Description( { text, learnMoreLink } ) {
+export default function Description( {
+	className = 'googlesitekit-publisher-win__desc',
+	text,
+	learnMoreLink,
+} ) {
 	return (
-		<div className="googlesitekit-publisher-win__desc">
+		<div className={ className }>
```
Comment (tofumatt) [inline assets/js/googlesitekit/notifications/components/common/Description.js:28]: Same here; let's use `classnames()`

File: assets/js/googlesitekit/notifications/components/layout/NotificationWithSVG.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	useBreakpoint,
+} from '../../../../hooks/useBreakpoint';
+import { Cell, Grid, Row } from '../../../../material-components';
+
+export default function NotificationWithSVG( {
+	id,
+	title,
+	description,
+	actions,
+	SVG,
```
Comment (tofumatt) [inline assets/js/googlesitekit/notifications/components/layout/NotificationWithSVG.js:37]: Capital prop names are a bit odd. I'm guessing it's done so you can use `<SVG />` below…

Looks like [we do this elsewhere too](https://github.com/google/site-kit-wp/blob/4e8b12560bfc561dae3d6d7c3419ad2d68a1ba11/assets/js/googlesitekit/notifications/components/layout/NotificationWithSmallSVG.js#L27), so no need to change it, but I guess leaving my comment here so it's recognised 😆

File: assets/sass/components/sign-in-with-google/_googlesitekit-sign-in-with-google-setup-cta-banner.scss [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/**
+ * Sign In With Google Setup CTA Banner styles.
```
Comment (tofumatt) [inline assets/sass/components/sign-in-with-google/_googlesitekit-sign-in-with-google-setup-cta-banner.scss:2]:
```suggestion
 * Sign in with Google Setup CTA Banner styles.
```

**Review by tofumatt — CHANGES_REQUESTED** [review]
A few questions/small comments but otherwise looks good 👍🏻

File: assets/js/components/consent-mode/ConsentModeSetupCTAWidget.js
Comment (jimmymadon) [reply assets/js/components/consent-mode/ConsentModeSetupCTAWidget.js:1]: So to reuse the same `<Description>` component within the new layout - I had to make it similar to the way it is being used in existing layouts. This meant wrapping the `p` in a `div`. Alternatively, I could've created YET another set of unique styles for this banner compared to the others which would eventually have to be refactored in the future. So I just decided to make minor modifications to the `ConsentModeSetupCTAWidget` as well as `AdsModuleSetupCTAWidget` as these use the same description styles.

File: assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js (hunk 1 above)
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/components/common/ActionsCTALinkDismiss.js:46]: I understand, but here, the class name is essentially a flag for the version we need to use for this component. Unfortunately, `googlesitekit-publisher-win__actions` and `googlesitekit-setup-cta-banner__actions-wrapper` are conflicting with each other - so we can either apply one or the other. If we were to use `classnames`, we would have to create another flag like `type` or something solely for styling. Type could then be `banner-notification` or `setup-cta`. And then we'd have to switch the classname based on the type. Ideally, we should be refactoring the legacy banner notifications (viz. all classes with `googlesitekit-publisher-win*`) soon.

So is this worth add the whole type flag, etc or should I leave this as is?

File: assets/js/googlesitekit/notifications/components/common/Description.js (hunk 2 above)
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/components/common/Description.js:28]: Same reasoning as above.

File: assets/js/googlesitekit/notifications/components/layout/NotificationWithSVG.js (hunk 3 above)
Comment (jimmymadon) [reply assets/js/googlesitekit/notifications/components/layout/NotificationWithSVG.js:37]: Hmm, this just clearly screams that we need a component as a prop here - similar to how we use pass `Widget`, `WidgetNull`, `Notification`, etc as well.

File: tests/backstop/reference/google-site-kit_ConsentMode_ConsentModeSetupCTAWidget_Default_0_document_2_large.png
Comment (jimmymadon) [inline tests/backstop/reference/google-site-kit_ConsentMode_ConsentModeSetupCTAWidget_Default_0_document_2_large.png:1]: @tofumatt Just to clarify, these changes are actually standardising the fonts of the description for existing components on certain viewports.

**Review by tofumatt — APPROVED** [review]

**Comment by tofumatt** [comment]
Thanks for the explanations/clarifications, this is good-to-go 👍🏻

**Pull request merged**